pkg/controllers/resources/pods/validatePodSecurity.go

```diff
@@ -0,0 +1,110 @@
+package pods
+
+import (
+	"context"
+	"fmt"
+
+	admissionv1 "k8s.io/api/admission/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	kubeinformers "k8s.io/client-go/informers"
+	"k8s.io/pod-security-admission/admission"
+	admissionapi "k8s.io/pod-security-admission/admission/api"
+	"k8s.io/pod-security-admission/api"
+	"k8s.io/pod-security-admission/metrics"
+	"k8s.io/pod-security-admission/policy"
+)
+
+func validatePodSecurityStandards(ctx context.Context, pod *corev1.Pod, s *podSyncer) (*admissionv1.AdmissionResponse, error) {
+	// if _, ok := podsecurity.PodSecurityExemptNamespaces[pod.Namespace]; ok {
+	// 	return &admissionv1.AdmissionResponse{
+	// 		Allowed: true,
+	// 	}, nil
+	// }
+
+	version := api.LatestVersion()
+	podSecurityDefaults := admissionapi.PodSecurityDefaults{
+		Enforce:        s.podSecurityStandard,
+		EnforceVersion: version.String(),
+		Audit:          s.podSecurityStandard,
+		AuditVersion:   version.String(),
+		Warn:           s.podSecurityStandard,
+		WarnVersion:    version.String(),
+	}
+
+	informerFactory := kubeinformers.NewSharedInformerFactory(s.virtualClusterClient, 0 /* no resync */)
+	namespaceInformer := informerFactory.Core().V1().Namespaces()
+	namespaceLister := namespaceInformer.Lister()
```

Comment:
Please rewrite this so that we don't need an new informer here, rather use the ctx.VirtualClient for this (create a new struct that implements the namespace lister interface and just uses the crx.VirtualClient under the hood)

Reply:
Done. PTAL.